### PR TITLE
Blog's settings

### PR DESCRIPTION
**Pull request by vied12**
## Description

new endpoint `global_preferences` added.  
It give access to a collection of `{key:"xxx", value: "yyy"}` objects.

When a Blog is created, all these settings are set to its new `blog_preferences` attribute in a `dict` format.

## Changelog

- [x] `Blog` model has now a `blog_preferences` attribute which contains the blog settings (based on globalPreference)
- [x] new `BlogSettings` view added with dedicated controller
- [x] a prepopulate system that provide initial database content (for `globalPreference` for instance)
- [x] implement the `get_blogbal_prefs` method
- [x] write tests for blog's `blog_preferences`

## TODO
- [ ] connect the view to the api

### CODE REVIEW

File: server/liveblog/global_preferences.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def init_app(app):
+    endpoint_name = _preferences_key
+    service = GlobalPreferencesService(endpoint_name, backend=get_backend())
+    GlobalPreferencesResource(endpoint_name, app=app, service=service)
+    superdesk.intrinsic_privilege(resource_name=endpoint_name, method=['PATCH'])
+
+
+class GlobalPreferencesResource(PackageResource):
```
Comment (ioanpocol) [inline server/liveblog/global_preferences.py]: Why did you use here PackageResource?

File: server/liveblog/global_preferences.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    }
+    schema = {
+        _general_preferences_key: {'type': 'dict', 'required': True}
+    }
+    resource_methods = ['GET', 'POST']
+    item_methods = ['GET', 'PATCH']
+
+
+class GlobalPreferencesService(PackageService):
```
Comment (ioanpocol) [inline server/liveblog/global_preferences.py]: Why did you use here PackageService?

File: server/liveblog/languages/__init__.py [hunk 3]
```diff
@@ -1,8 +0,0 @@
-import superdesk
```
Comment (ioanpocol) [inline server/liveblog/languages/__init__.py]: You should keep this endpoint

File: server/liveblog/themes/themes.py [hunk 4]
```diff
@@ -1,34 +0,0 @@
-from superdesk.resource import Resource
```
Comment (ioanpocol) [inline server/liveblog/themes/themes.py]: You should keep this endpoint

File: server/liveblog/global_preferences.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        'key': {'type': 'str', 'required': True},
+        'value': {'type': 'str', 'required': True}
+    }
+    resource_methods = ['GET', 'POST']
+    item_methods = ['GET', 'PATCH']
+
+
+class GlobalPreferencesService(BaseService):
+    def get_global_prefs(self):
+        return {'language': 'en', 'theme': 'theme1'}
```
Comment (ioanpocol) [inline server/liveblog/global_preferences.py]: You should get from db the saved values for "language" and "theme" keys

File: server/liveblog/global_preferences.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    def get(self, req, lookup):
```
Comment (ioanpocol) [inline server/liveblog/global_preferences.py]: This is the exact implementation from BaseService, so not needed to overwrite again.

File: server/features/steps/steps.py [hunk 7]
```diff
@@ -984,6 +984,14 @@ def get_default_prefs(context):
     assert_equal(response_data['user_preferences'], default_user_preferences)
 
 
+@then('we get default blog preferences')
```
Comment (ioanpocol) [inline server/features/steps/steps.py]: Can be deleted, not needed and to work need changes

File: server/liveblog/prepopulate.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8; -*-
```
Comment (ioanpocol) [inline server/liveblog/prepopulate.py]: It is better to add prepopulate related changes to liveblog fork of superdesk:
https://github.com/liveblog/superdesk

**Comment by vladnicoara** [comment]
except the .py part that I don't know too much about... LGTM

**Comment by ioanpocol** [comment]
:+1:

File: .gitignore [hunk 9]
```diff
@@ -10,6 +10,8 @@ __pycache__/
 # Distribution / packaging
 .Python
 env/
+venv/
```
Comment (nistormihai) [inline .gitignore]: are you using pyvenv vs virtualenv?
isn't this contained in docker? or you got outside of docker :8ball:

File: client/app/scripts/liveblog-edit/directives.js [hunk 10]
```diff
@@ -214,6 +214,16 @@ define([
                 }
             };
         })
+        .directive('selectAllOnClick', [function() {
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/directives.js]: IMHO select-all isn't a very good name for this, much better just selectOnCick or selectTextOnClick

File: client/app/scripts/liveblog-edit/module.js [hunk 11]
```diff
@@ -90,7 +89,7 @@ define([
             },
             saveAsDraft: function() {
                 notify.info(gettext('Saving draft'));
-                postsService.saveDraft(current_blog_id, current_post, getItemsFromEditor()).then(function(post) {
+                postsService.saveDraft(blog._id, current_post, getItemsFromEditor()).then(function(post) {
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/module.js]: good :100:

File: client/app/scripts/liveblog-edit/module.js [hunk 12]
```diff
@@ -163,6 +162,21 @@ define([
...
     }
 
+    BlogSettingsController.$inject = ['blog', 'api'];
+    function BlogSettingsController(blog, api) {
+        var vm = this;
+        // set scope
+        angular.extend(vm, {
+            blog: blog,
+            blogSettings: {},
+            languages: [] // loaded after
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/module.js]: comment irrelevant.

File: server/Dockerfile [hunk 13]
```diff
@@ -3,13 +3,7 @@ FROM ubuntu:trusty
...
 # install python3 and build dependencies for c modules
 # update pip and distribute
-RUN apt-get update && \
-DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
-python3 python3-dev python3-pip \
-build-essential libffi-dev git mercurial \
-libtiff5-dev libjpeg8-dev zlib1g-dev \
-libfreetype6-dev liblcms2-dev libwebp-dev \
-curl
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends python3 python3-dev python3-pip build-essential libffi-dev git mercurial libtiff5-dev libjpeg8-dev zlib1g-dev libfreetype6-dev liblcms2-dev libwebp-dev curl
```
Comment (nistormihai) [inline server/Dockerfile]: better looked the deleted one.

File: server/apps/prepopulate/app_prepopulate_data.json [hunk 14]
```diff
@@ -16,5 +16,36 @@
     "data": {
         "name": "Editor"
     }
+},
+{
+    "resource": "global_preferences",
+    "data": {
+        "key": "language",
+        "value": "fr"
```
Comment (nistormihai) [inline server/apps/prepopulate/app_prepopulate_data.json]: why fr? :smile:

File: server/fig.yml [hunk 15]
```diff
@@ -1,41 +0,0 @@
-db:
```
Comment (nistormihai) [inline server/fig.yml]: don't we need fig anymore? there is still used in README.md
please remove it from there also if it is the case.

File: server/Dockerfile (hunk 13 above)
Comment (vied12) [inline server/Dockerfile]: I agree but the deleted one was bad interpreted. The ` \` was removed without spaces between words.

File: server/apps/prepopulate/app_prepopulate_data.json (hunk 14 above)
Comment (vied12) [inline server/apps/prepopulate/app_prepopulate_data.json]: I just tried :trollface:

File: server/fig.yml (hunk 15 above)
Comment (vied12) [inline server/fig.yml]: Good point, I have to update the README.md.
We don't use fig anymore but docker-compose. And the template file is in `docker/`.
Anyway we need a better README, I did this one 5s before the initial commit on the new repo.

File: server/Dockerfile (hunk 13 above)
Comment (vied12) [inline server/Dockerfile]: Fixed, maybe by using the latest Docker

**Comment by nistormihai** [comment]
LGTM, but please update from master.

**Pull request merged**